src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/ResetOidcSecretCommandHandler.php

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Application\CommandHandler\Entity;
+
+use Ramsey\Uuid\Uuid;
+use Surfnet\ServiceProviderDashboard\Application\Command\Entity\ResetOidcSecretCommand;
+use Surfnet\ServiceProviderDashboard\Application\Service\LoadEntityService;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+use Surfnet\ServiceProviderDashboard\Application\CommandHandler\CommandHandler;
+use Surfnet\ServiceProviderDashboard\Application\Exception\EntityNotFoundException;
+use Surfnet\ServiceProviderDashboard\Application\Exception\InvalidArgumentException;
+use Surfnet\ServiceProviderDashboard\Domain\Repository\EntityRepository;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Secret;
+
+class ResetOidcSecretCommandHandler implements CommandHandler
+{
+    /**
+     * @var EntityRepository
+     */
+    private $repository;
+    /**
+     * @var LoadEntityService
+     */
+    private $loadEntityService;
+
+    /**
+     * @param EntityRepository $repository
+     * @param LoadEntityService $loadEntityService
+     */
+    public function __construct(EntityRepository $repository, LoadEntityService $loadEntityService)
+    {
+        $this->repository = $repository;
+        $this->loadEntityService = $loadEntityService;
+    }
+
+    /**
+     * @param ResetOidcSecretCommand $command
+     * @throws EntityNotFoundException
+     * @throws InvalidArgumentException
+     * @throws \Surfnet\ServiceProviderDashboard\Infrastructure\Manage\Exception\QueryServiceProviderException
+     */
+    public function handle(ResetOidcSecretCommand $command)
+    {
+        $entity = $this->loadEntityService->load(
+            null,
+            $command->getManageId(),
+            $command->getService(),
+            $command->getEnvironment(),
+            $command->getEnvironment()
+        );
+
+        if (!$entity) {
+            throw new EntityNotFoundException('The requested entity could not be found');
+        }
+
+        if ($entity->getProtocol() !== Entity::TYPE_OPENID_CONNECT) {
+            throw new EntityNotFoundException('The requested entity could be found, invalid protocol');
+        }
+
+        if ($entity->getStatus() !== Entity::STATE_PUBLISHED) {
+            throw new EntityNotFoundException('The requested entity could be found, invalid state');
+        }
+
+        $id = (string) Uuid::uuid1();
+        if (!$this->repository->isUnique($id)) {
+            throw new InvalidArgumentException(
+                'The id that was generated for the entity was not unique, please try again'
+            );
+        }
+
+        $secret = new Secret(Entity::OIDC_SECRET_LENGTH);
+
+        $entity->setId($id);
```

Comment:
This violates CQRS. You should provide the ID with the command. And maybe the uuid1() should return a unique id that you don't have to worry about (as in not checking it again)

Reply:
This should go for line 94

Reply:
Fixed